extensions/smallrye-context-propagation/runtime/src/main/java/io/quarkus/smallrye/context/runtime/context/provider/ArcContextProvider.java

```diff
@@ -0,0 +1,129 @@
+                requestContext.deactivate();
+                requestContext.activate(instancesToPropagate);
+                controller = () -> {
+                    // clean up, reactivate context with previous values
+                    requestContext.deactivate();
+                    requestContext.activate(instancesToRestore);
+                };
+            } else {
+                // context not active, activate and pass it new instance, deactivate afterwards
+                requestContext.activate(instancesToPropagate);
+                controller = () -> {
+                    requestContext.deactivate();
+                };
+            }
+            return controller;
+        };
+    }
+
+    @Override
+    public ThreadContextSnapshot clearedContext(Map<String, String> map) {
+        // note that by cleared we mean that we still activate context if need be, just leave the contents blank
+        if (!isCdiAvailable()) {
+            //return null as per docs to state that propagation of this context is not supported
+            return null;
+        }
+
+        ArcContainer arc = Arc.container();
+        if (!isContextActiveOnThisThread(arc)) {
+            // request context not active, nothing to propagate, return no-op
+            return NOOP_SNAPSHOT;
+        }
+
+        return () -> {
+            ThreadContextController controller;
+            ManagedContext requestContext = arc.requestContext();
+            // this is executed on another thread, context can but doesn't need to be active here
+            if (isContextActiveOnThisThread(arc)) {
+                // context active, store current state, start blank context anew and restore state afterwards
+                Collection<ContextInstanceHandle<?>> instancesToRestore = requestContext.getAll();
+                requestContext.deactivate();
+                requestContext.activate();
+                controller = () -> {
+                    // clean up, reactivate context with previous values
+                    requestContext.deactivate();
+                    requestContext.activate(instancesToRestore);
+                };
+            } else {
+                // context not active, activate blank one, deactivate afterwards
+                requestContext.activate();
+                controller = () -> {
+                    requestContext.deactivate();
+                };
+            }
+            return controller;
+        };
+    }
+
+    @Override
+    public String getThreadContextType() {
+        return ThreadContext.CDI;
+    }
+
+    /**
+     * Checks if CDI is available within the application by using {@code CDI.current()}.
+     * If an exception is thrown, it is suppressed and false is returns, otherwise true is returned.
+     *
+     * @return true if CDI can be used, false otherwise
+     */
+    private boolean isCdiAvailable() {
+        try {
+            return CDI.current() != null;
```

Comment:
After discussing this with Martin, we decides its ineffective and we should instead introduce and use `ArcContainer.isRunning()`.

Reply:
@FroMage Apparently I am not allowed to contribute to your existing branch or I am doing it wrong. Here is the commit you need to cherry-pick here - https://github.com/manovotn/quarkus/commit/7a54d8476e7cb4a571c3245914e257919f528161

Reply:
Just added you as colab, you should be able to push to this branch now.

Reply:
Ok, will try

Reply:
Pushed the commit here